src/Migrations.php

```diff
@@ -0,0 +1,206 @@
+     * @param array $options Options to pass to the command
+     * Available options are :
+     *
+     * - `format` Format to output the response. Can be 'json'
+     * - `connection` The datasource connection to use
+     * - `source` The folder where migrations are in
+     * - `plugin` The plugin containing the migrations
+     *
+     * @return array The migrations list and their statuses
+     */
+    public function status($options = [])
+    {
+        $input = $this->getInput('Status', $options);
+        $params = ['default', $input->getOption('format')];
+
+        return $this->run('printStatus', $params, $input);
+    }
+
+    /**
+     * Migrates available migrations
+     *
+     * @param array $options Options to pass to the command
+     * Available options are :
+     *
+     * - `target` The version number to migrate to. If not provided, will migrate
+     * everything it can
+     * - `connection` The datasource connection to use
+     * - `source` The folder where migrations are in
+     * - `plugin` The plugin containing the migrations
+     *
+     * @return bool Success
+     */
+    public function migrate($options = [])
+    {
+        $input = $this->getInput('Migrate', $options);
+        $params = ['default', $input->getOption('target')];
+
+        try {
+            $this->run('migrate', $params, $input);
+            return true;
+        } catch (\Exception $e) {
+            $this->lastError = $e->getMessage();
+            return false;
+        }
+    }
+
+    /**
+     * Rollbacks migrations
+     *
+     * @param array $options Options to pass to the command
+     * Available options are :
+     *
+     * - `target` The version number to migrate to. If not provided, will only migrate
+     * the last migrations registered in the phinx log
+     * - `connection` The datasource connection to use
+     * - `source` The folder where migrations are in
+     * - `plugin` The plugin containing the migrations
+     *
+     * @return bool Success
+     */
+    public function rollback($options = [])
+    {
+        $input = $this->getInput('Rollback', $options);
+        $params = ['default', $input->getOption('target')];
+
+        try {
+            $this->run('rollback', $params, $input);
+            return true;
+        } catch (\Exception $e) {
+            $this->lastError = $e->getMessage();
+            return false;
+        }
+    }
+
+    /**
+     * Runs the method needed to execute and return
+     *
+     * @param string $method Manager method to call
+     * @param array $params Manager params to pass
+     * @param \Symfony\Component\Console\Input\InputInterface InputInterface needed for the
+     * Manager to properly run
+     *
+     * @return mixed The result of the CakeManager::$method() call
+     */
+    protected function run($method, $params, $input)
+    {
+        $this->setInput($input);
+        $config = $this->getConfig();
+        $manager = new CakeManager($config, $this->output);
```

Comment:
Check if the same instance can be used and the ConfigInterface instance swapped with a method from the original phinx Manager class